message/infrastructure/amqp/topology_builder.go

```diff
@@ -0,0 +1,54 @@
+package amqp
+
+import (
+	"github.com/ThreeDotsLabs/watermill"
+	"github.com/pkg/errors"
+	"github.com/streadway/amqp"
+)
+
+type topologyBuilder interface {
+	buildTopology(channel *amqp.Channel, queueName string, logFields watermill.LogFields, exchangeName string, exchangeDeclarer exchangeDeclarer) error
```

Comment:
what do you think about adding exchangeDeclarer `DeclareExchange(channel *amqp.Channel, exchangeName string) error) ?` to the interface?

Reply:
yeah, definitely good idea